code/iaas/allocator/src/main/java/io/cattle/platform/allocator/service/AllocatorServiceImpl.java

```diff
@@ -87,13 +87,19 @@ public void mergeLabels(Map<String, String> srcMap, Map<String, String> destMap)
             return;
         }
         for (Map.Entry<String, String> entry : srcMap.entrySet()) {
-            String destValue = destMap.get(entry.getKey());
-            if (StringUtils.isEmpty(destValue)) {
-                destMap.put(entry.getKey(), entry.getValue());
-            } else if (StringUtils.isEmpty(entry.getValue())) {
-                continue;
+            if (entry.getKey().startsWith("io.rancher.scheduler.affinity")) {
```

Comment:
Should we introduce some variable for the label name, instead of hardcoding? In case we decide to change it in the future